src/entities/database.js

```diff
@@ -284,6 +284,43 @@ class DB implements Database {
     )
   }
 
+  async queryLoosePendingTxs() {
+    const snapshotsTable = 'pending_snapshots'
+    const query = Q.sql.select().from(snapshotsTable)
+      .field('tx_hash')
+      .where('block_height = ?', Q.sql.select().from(snapshotsTable)
+        .field('MAX(block_height)'))
+      .union(Q.sql.select().from('txs')
+        .field('hash')
+        .where('tx_state = ?', TX_STATUS.TX_PENDING_STATUS)
+        .where('NOT EXISTS ?', Q.sql.select().from(snapshotsTable)
+          .field('1')
+          .where('tx_hash = hash')))
+      .toString()
+    const dbRes = await this.getConn().query(query)
+    this.#logger.debug('queryLoosePendingTxs:', query, dbRes)
+    return _.map(dbRes.rows, 'tx_hash')
+  }
+
+  async storeNewPendingSnapshot(block: Block) {
+    const loosePendingSet = await this.queryLoosePendingTxs()
```

Comment:
Here `loosePendingSet` should also be renamed into just `pendingSet` because it includes both loose and committed pending txs.